31-plotting_sample_size_biological_context_coverage.Rmd

```diff
@@ -0,0 +1,461 @@
+  recount2.plot <- recount2.df %>%
+    ggplot(aes_string(x = "condition", y = y.variable.name)) +
+    geom_point(fill = "#0000FF", shape = 23, size = 4) +
+    custom_theme() +
+    ylim(eval(parse(text = y.limits))) +
+    labs(y = NULL, title = "MultiPLIER")
+  
+  # return a list of the three plots
+  return(list("size" = size.plot,
+              "context" = context.plot,
+              "recount2" = recount2.plot))
+}
+
+```
+
+```{r}
+MultiPanelWrapper <- function(size.plot, context.plot, recount2.plot, 
+                              plot.title, output.file) {
+  # given the plots for the sample size experiment, the biological context
+  # experiment and the full MultiPLIER model, respectively, make a multipanel
+  # plot titled with the plot.title argument and saved as an 11"W x 5"H plot
+  # in the location specified by output.file
+  
+  # for the three conditions, make a 3 column multipanel plot
+  panels <- cowplot::plot_grid(size.plot, context.plot, recount2.plot, 
+                               align = "h", 
+                               # the size plot will have the y-axis label
+                               # and the MultiPLIER panel should be quite s
+                               # small
+                               rel_widths = c(1.125, 1, 0.375), 
+                               ncol = 3)
+  
+  # the title "panel"
+  p.title <- cowplot::ggdraw() + 
+    cowplot::draw_label(plot.title, fontface = "bold", 
+                        size = 20)
+  
+  # combine to get the final plot
+  final.plot <- cowplot::plot_grid(p.title, panels, ncol = 1, 
+                                   rel_heights = c(0.1, 1))
+  
+  # save 11"W x 5"H file (the type of file is determined by the file extension
+  # provided in output.file)
+  ggsave(output.file, plot = final.plot, width = 11, height = 5)
+  
+}
+```
+
+#### Directory setup
+
+```{r}
+# plot and result directory setup for this notebook
+plot.dir <- file.path("plots", "31")
+dir.create(plot.dir, recursive = TRUE, showWarnings = FALSE)
+results.dir <- file.path("results", "31")
+dir.create(results.dir, recursive = TRUE, showWarnings = FALSE)
+```
+
+## Read in data
+
+For both the biological context and sample size evaluations, the results we
+will be plotting are from `30-evaluate_sample_size_and_biological_context`.
+
+### Biological context
+
+```{r}
+# pathway coverage and proportion of LVs associated with pathways
+context.coverage.file <- file.path("results", "30", 
+                                   "biological_context_pathway_coverage.tsv")
+context.coverage.df <- readr::read_tsv(context.coverage.file) %>%
+  # clean up the context information for plotting
+  dplyr::mutate(biological_context = gsub("_", " ", biological_context)) %>%
+  # reorder by the sample size (low -> high)
+  dplyr::mutate(biological_context = factor(biological_context,
+                                            levels = c("blood", "cancer",
```

Comment:
maybe the levels can be split out and saved in a new variable - i see it is repeated in line 188